src/components/ExchangeSummary.tsx

```diff
@@ -126,13 +127,27 @@ const ExchangeSummary = (props: {
           </div>
         )}
       </TableDataRightAlign>
+      <TableDataRightAlign>
+        {tradeLoading ? (
+          <CircularProgress />
+        ) : (
+          <div>{Math.ceil(parseInt(props.desiredAmount) / maxHalfTradeToken)}</div>
```

Comment:
I'd suggest breaking this out into it's own function, since the logic is repeated. 

Also, when the `maxHalfTradeToken`, or other similar value, is zero, the result is infinity. Can we default to returning `0` when `maxHalfTradeToken == 0`?

<img width="365" alt="Screen Shot 2021-10-29 at 8 13 06 PM" src="https://user-images.githubusercontent.com/7647623/139513373-cd0cabdd-35e0-4302-9736-822d0d917445.png">

Reply:
Definitely! And definitely! I'm also breaking the `<TableDataRightAlign />`s into a function because they feel too repetitive but can absolutely leave them alone instead.

Reply:
example: 
```
<TableDataRightAlign>
        {isLoading ? (
          <CircularProgress />
        ) : (
          <div>
            {isError
              ? 'Error'
              : value}
          </div>
        )}
      </TableDataRightAlign>
```
to
```
      {renderCustomTableData(halfTradeLoading, formatUSD(maxHalfTradeToken), halfTradeError)}
```